src/stamina/_core.py

```diff
@@ -51,10 +60,14 @@ async def _smart_sleep(delay: float) -> None:
 
 T = TypeVar("T")
 P = ParamSpec("P")
+# for backwards compatibility with Python<3.10
+On = Union[
+    Type[Exception], Tuple[Type[Exception], ...], Callable[[Exception], bool]
+]
```

Comment:
I'm not a big fan of the name. `on: On` doesn't convey any meaning to the reader of the code.

My own inclination would be to try and give some meaning to the name - e.g. `on: exc_or_predicate`. It's not perfect, but I think it's probably better than conveying no meaning in the name.

Reply:
Renamed to `ExcOrPredicate`